server/web_ui/src/components/totpdisplay.rs

```diff
@@ -0,0 +1,322 @@
+#![allow(warnings)]
+
+#[cfg(debug_assertions)]
+use std::ops::Add;
+
+use gloo::console;
+use gloo::console::console;
+use gloo_timers::callback::{Interval, Timeout};
+use js_sys::Array;
+use kanidm_proto::internal::{IdentifyUserRequest, IdentifyUserResponse};
+use wasm_bindgen::JsValue;
+use wasm_timer::SystemTime;
+use yew::prelude::*;
+
+use crate::constants::{CLASS_DIV_LOGIN_BUTTON, ID_IDENTITY_VERIFICATION_SYSTEM_TOTP_MODAL};
+use crate::utils::document;
+use crate::views::identityverification::{
+    IdentifyUserState, IdentifyUserTransition, CORRUPT_STATE_ERROR,
+};
+use crate::{do_request, utils, RequestMethod};
+
+static TIME_STEP: u128 = 30;
```

Comment:
This should be returned from the server when you get the TOTP so we can just the TOTP window server side and then the ui reflects that.